### ISSUE TITLE
Datepicker whole date is read when switching between items in month/year view

### ISSUE THREAD

**Issue by mparvanov** [telerik/kendo-ui-core#5498]
### Bug report

In the month/year view, if I change the selection, the selected date (from the original view before "zooming out" into the month/year view) is read out.

### Reproduction of the problem
1. Open https://demos.telerik.com/kendo-ui/datepicker/rangeselection 
1. Start a screenreader (NVDA or JAWS).
1. Focus on the calendar, open it.
1. Change the view to the month/year view by using Ctrl+Arrow Up.
1. The focus is now on the month/year corresponding to the current date.
1. Change the selection by using one of the arrow keys.

### Current behavior
No matter the current selection, always the (whole) current date is read out.

### Expected/desired behavior
The currently selected month/year should be read out only

Public item:
https://feedback.telerik.com/kendo-vue-ui/1446886-datepicker-accessibility-issue-in-month-year-view

### Environment

* **Kendo UI version:** 2019.3.1023
* **Screenreader (always initial configuration)**:
NVDA 2019.1.1
JAWS 2019.1906.10 ILM

**Comment by Alex-Bubblemaster** [comment]
As a workaround, the [`ARIATemplate`](https://docs.telerik.com/kendo-ui/api/javascript/ui/datepicker/configuration/ariatemplate) could be used.

[Dojo](https://dojo.telerik.com/@bubblemaster/oQUWAcoM/2)

```
 ARIATemplate: "Current focused date is #= this.dateView.calendar.view().name === 'month' ? kendo.toString(data.current, 'D'): kendo.toString(data.current, 'MMM yyyy') #",
```

**Comment by ag-petrov** [comment]
@dmanova fixed for all pickers

**Comment by dmanova** [comment]
_Notes from internal testing:_
In DateRangePicker seems the focus is kept on the date input and the readers are confused when reading the dates from the calendar. Also when removing the focus from the input the aria-label is not generated. Check the attached files.

[DataRangePicker-Chrome-NVDA-Reading.zip](https://github.com/telerik/kendo-ui-core/files/6061443/DataRangePicker-Chrome-NVDA-Reading.zip)

**Comment by dmanova** [comment]
Some readers also read additional information or word "blank" that is unnecessary, but despite of this the correct current date/month/year is read now, by JAWS and NVDA.

**Comment by dmanova** [comment]
It seems the fix is introducing some regression. Check the example with the workaround in the comment above (https://github.com/telerik/kendo-ui-core/issues/5498#issuecomment-587009476). It throws JS Error now.
